src/ServiceControl.AcceptanceTests/Recoverability/ExternalIntegration/When_a_failed_message_is_resolved_by_retry.cs

```diff
@@ -0,0 +1,158 @@
+    using NServiceBus.Settings;
+    using ServiceBus.Management.Infrastructure.Settings;
+    using NUnit.Framework;
+    using ServiceControl.Contracts;
+    using ServiceControl.MessageFailures;
+    using TestSupport;
+    using TestSupport.EndpointTemplates;
+    using Newtonsoft.Json;
+
+    class When_a_failed_message_is_resolved_by_retry : AcceptanceTest
+    {
+        [Test]
+        public async Task Should_publish_notification()
+        {
+            CustomConfiguration = config => config.OnEndpointSubscribed<MyContext>((s, ctx) =>
+            {
+                if (s.SubscriberReturnAddress.IndexOf("ExternalProcessor", StringComparison.OrdinalIgnoreCase) >= 0)
+                {
+                    ctx.ExternalProcessorSubscribed = true;
+                }
+            });
+
+            var context = await Define<MyContext>()
+                .WithEndpoint<Receiver>(b => b.When(c => c.ExternalProcessorSubscribed, async bus =>
+                {
+                    await bus.SendLocal<MyMessage>(m => m.MessageNumber = 1)
+                        .ConfigureAwait(false);
+                }).DoNotFailOnErrorMessages())
+                .WithEndpoint<ExternalProcessor>(b => b.When(async (bus, c) =>
+                {
+                    await bus.Subscribe<MessageFailureResolvedByRetry>();
+
+                    if (c.HasNativePubSubSupport)
+                    {
+                        c.ExternalProcessorSubscribed = true;
+                    }
+                }))
+                .Do("WaitUntilErrorsContainsFaildMessage", async ctx =>
+                {
+                    if (ctx.FailedMessageId == null)
+                    {
+                        return false;
+                    }
+
+                    FailedMessage failedMessage = await this.TryGet<FailedMessage>($"/api/errors/{ctx.FailedMessageId}");
+                    if (failedMessage == null)
+                    {
+                        return false;
+                    }
+
+                    return true;
+                })
+                .Do("Retry", async ctx =>
+                {
+                    ctx.AboutToSendRetry = true;
+                    ctx.FailProcessing = false;
+                    await this.Post<object>($"/api/errors/{ctx.FailedMessageId}/retry");
+                })
+                .Do("EnsureRetried", async ctx =>
+                {
+                    return await this.TryGet<FailedMessage>($"/api/errors/{ctx.FailedMessageId}",
+                        e => e.Status == FailedMessageStatus.Resolved);
+                })
+                .Done(ctx => ctx.EventDelivered) //Done when sequence is finished
+                .Run();
+
+            var deserializedEvent = JsonConvert.DeserializeObject<MessageFailureResolvedByRetry>(context.Event);
+            Assert.IsTrue(deserializedEvent.FailedMessageId == context.FailedMessageId);
+        }
+
+        public class Receiver : EndpointConfigurationBuilder
```

Comment:
Could we move this to a test base class to prevent duplication?

Reply:
@tmasternak you are refactoring this now, so ignoring this for now